src/EVM/Expr.hs

```diff
@@ -203,11 +203,12 @@ sar = op2 SAR (\x y ->
 
 peq :: (Typeable a) => Expr a -> Expr a -> Prop
 peq (Lit x) (Lit y) = PBool (x == y)
-peq a@(Lit _) b = PEq a b
-peq a b@(Lit _) = PEq b a -- we always put concrete values on LHS
+peq (LitByte x) (LitByte y) = PBool (x == y)
+peq (ConcreteBuf x) (ConcreteBuf y) = PBool (x == y)
 peq a b
   | a == b = PBool True
-  | otherwise = PEq a b
+  | otherwise = let args = sort [a, b]
```

Comment:
Sort prefers Lit above all else, so it will put Lit as 1st. But it will also order everything else, which is what we want.